streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java

```diff
@@ -52,12 +52,14 @@
      *
      * @param reducer           the instance of {@link Reducer}
      * @param windows           the specification of the aggregation {@link Windows}
+     * @param storeName         the name of the state store created from this operation
```

Comment:
Would it make sense to elaborate here why the user must provide this name at all?  It is IMHO counter-intuitive to require this string parameter, so let's explain why it is needed.  Also, we may want to explain how the user can actually leverage/re-use the state store that was named here.